pkg/morph/deploy/group.go

```diff
@@ -0,0 +1,345 @@
+						domain, _actor.Sender(), prm.systemEmail, nnsRefresh, nnsRetry, nnsExpire, nnsMinimum)
+					if err != nil {
+						vubs[0] = 0
+						mDomainsToVubs[domain] = vubs
+						if isErrNotEnoughGAS(err) {
+							l.Info("not enough GAS to register domain in the NNS, will try again later")
+						} else {
+							l.Error("failed to send transaction registering domain in the NNS, will try again later", zap.Error(err))
+						}
+						return
+					}
+
+					vubs[0] = vub
+					mDomainsToVubs[domain] = vubs
+
+					l.Info("transaction registering domain in the NNS has been successfully sent, will wait for the outcome")
+
+					continue
+				} else if !errors.Is(err, errMissingDomainRecord) {
+					l.Error("failed to lookup NNS domain record, will try again later", zap.Error(err))
+					return
+				}
+
+				l.Info("missing record of the NNS domain, needed to be set")
+
+				vubs, ok := mDomainsToVubs[domain]
+				if ok && vubs[1] > 0 {
+					l.Info("transaction setting NNS domain record was sent earlier, checking relevance...")
+
+					if cur := monitor.currentHeight(); cur <= vubs[1] {
+						l.Info("previously sent transaction setting NNS domain record may still be relevant, will wait for the outcome",
+							zap.Uint32("current height", cur), zap.Uint32("retry after height", vubs[1]))
+						return
+					}
+
+					l.Info("previously sent transaction setting NNS domain record expired without side-effect")
+				}
+
+				l.Info("sharing encrypted committee group key with the committee member...")
+
+				keyCipher, err := encryptSharedPrivateKey(committeeGroupKey, prm.localAcc.PrivateKey(), prm.committee[i])
+				if err != nil {
+					l.Error("failed to encrypt committee group key to share with the committee member, will try again later",
+						zap.Error(err))
+					return
+				}
+
+				l.Info("sending new transaction setting domain record in the NNS...")
+
+				_, vub, err := _actor.SendCall(prm.nnsOnChainAddress, methodNNSAddRecord,
+					domain, int64(nns.TXT), keyCipher)
+				if err != nil {
+					vubs[1] = 0
+					mDomainsToVubs[domain] = vubs
+					if isErrNotEnoughGAS(err) {
+						l.Info("not enough GAS to set NNS domain record, will try again later")
+					} else {
+						l.Error("failed to send transaction setting NNS domain record, will try again later", zap.Error(err))
+					}
+					return
+				}
+
+				vubs[1] = vub
+				mDomainsToVubs[domain] = vubs
+
+				l.Info("transaction setting NNS domain record has been successfully sent, will wait for the outcome")
+
+				continue
+			}
+
+			l.Info("committee group key is shared with the committee member in NNS domain record")
+		}
+	}, nil
+}
+
+// encryptSharedPrivateKey encrypts private key using provided coder's private
+// key to be decrypted using decoder's private key. Inverse operation to
+// decryptSharedPrivateKey.
+func encryptSharedPrivateKey(sharedPrivKey, coderPrivKey *keys.PrivateKey, decoderPubKey *keys.PublicKey) (string, error) {
```

Comment:
I'd probably encrypt the private key with a symmetric cipher and then share this cipher's key via ECDH encryption, rather than encrypting the key directly with per-node symmetric key. But probably we can leave it as is until neo-project/proposals#161 makes it much simpler.

Reply:
lets wait for the proposal to be accepted, created https://github.com/nspcc-dev/neofs-node/issues/2405